traits/tests/test_on_trait_change.py

```diff
@@ -0,0 +1,64 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+
+from traits.api import HasTraits, Int, on_trait_change, TraitNotificationError
+
+
+class BaseTestClass(HasTraits):
+    foo = Int()
+    bar = Int()
+    baz = Int()
+    output_name = None
+    output_old = None
+    output_new = None
+
+
+class TestOnTraitChangeNotification(unittest.TestCase):
```

Comment:
This should probably be in `test_extended_trait_change` - that is where all the testing of the different listening patterns is, I think.